tests/test_adam.py

```diff
@@ -23,11 +37,12 @@ def adam(step, lr, param, grad, exp_avg, exp_avg_sq, beta1=0.9, beta2=0.999, eps
     step_size = lr / bias_correction1
     bias_correction2_sqrt = math.sqrt(bias_correction2)
     denom = (exp_avg_sq.sqrt() / bias_correction2_sqrt).add_(eps)
-    param.addcdiv_(exp_avg, denom, value=-step_size)
+    param.data = param.addcdiv(exp_avg, denom, value=-step_size).data
```

Comment:
Why?

Reply:
pytest doesn't allow in-place instructions for tensors that require gradient

Reply:
Wrap it with `torch.no_grad()`